main.go

```diff
@@ -115,17 +131,84 @@ func ProcessIncomingConnection(conn net.Conn) {
 
 		if decoder.Method == "icinga::Hello" {
 			log.Printf("SERVER: JsonRpcConnection: Received %q from the client\n", decoder.Method)
-		}
 
-		err = base.WriteNetStringToStream(conn, mes)
+			var encoder Encoder
+			encoder = Encoder{
+				JSONRPC: "2.0",
+				Method: "config::Update",
+				Params: Updates{
+					Update: map[string]ConfigInfo{
+						*zone: {},
+					},
+					Update_v2: map[string]ConfigInfo{
+						*zone: {},
+					},
+					Checksums: map[string]ConfigInfo{
+						*zone: {},
+					},
+				},
+			}
 
-		log.Printf("SERVER: Connection: Sending %d bytes message back", len(mes))
+			r := math.Intn(*files)
+			for i := 0; i < r; i++ {
+				file, err := GenerateFile("conf")
 
-		if err != nil {
-			log.Printf("SERVER: Connection: Sending data to the client Error: %s", err)
-			break
+				if err != nil {
+					log.Fatalf("SERVER: JsonRpcConnection: Error while generating new config files: %s", err)
+				}
+
+				encoder.Params.Update[*zone][file] = file
+
+				path, err := GenerateFile("conf")
+
+				if err != nil {
+					log.Fatalf("SERVER: JsonRpcConnection: Error while generating new config files: %s", err)
+				}
+
+				encoder.Params.Update_v2[*zone][path] = path
+
+				random, err := GenerateFile("conf")
+
+				if err != nil {
+					log.Fatalf("SERVER: JsonRpcConnection: Error while generating new config files: %s", err)
+				}
+
+				encoder.Params.Checksums[*zone][random] = random
+			}
+
+			mess, err := json.Marshal(&encoder)
+
+			if err != nil{
+				log.Fatalf("SERVER: JsonRpcConnection: Error while encoding JSRPC message. %s", err)
+			}
+
+			log.Printf("SERVER: JsonRpcConnection: Sending config::Update JSRPC message: %s", mess)
+
+			err = base.WriteNetStringToStream(conn, mess)
+
+			if err != nil {
+				log.Fatalf("SERVER: JsonRpcConnection: Error while sending json encoded JSRPC message: %s", err)
+			}
 		}
 	}
 
 	log.Println("SERVER: Connection: Socket closed")
 }
+
+func IsEmpty(val *string, desc string)  {
+	if *val == " " {
+		log.Fatalf("SERVER: Required command line '%s' argment is empty\n", desc)
+	}
+}
+
+func GenerateFile(suffix string) (string, error) {
+	r := math.Intn(1000000)
+	for i := 0; i < 10; i++ {
+		fname := fmt.Sprintf("%d.%s", r, suffix)
+		if _, err := os.Stat(fname); os.IsNotExist(err) {
```

Comment:
What's this condition for?

Reply:
To verify that there are no errors when creating the file.

Reply:
That's unnecessary.